src/groovy/com/recomdata/extensions/ExtensionsRegistry.groovy

```diff
@@ -1,13 +1,74 @@
 package com.recomdata.extensions
 
+import groovy.transform.CompileStatic
+
 /**
  * Date: 14-Jan-16
  * Time: 18:08
  */
 class ExtensionsRegistry {
     def analysisTabExtensions = []
+    List<Map<String, Object>> tabs = new LinkedList<>()
+
+    ExtensionsRegistry() {
+    }
 
     void registerAnalysisTabExtension(Map<String, Object> config = [:], String extensionId, String resourcesUrl, String bootstrapFunction) {
         analysisTabExtensions.add([extensionId: extensionId, resourcesUrl: resourcesUrl, bootstrapFunction: bootstrapFunction, config: config])
     }
+
+    void registerTab(Map<String, Object> config = [:], String id, String title, String controller) {
+        def tab = [id: id, title: title, controller: controller, insertBefore: config.insertBefore]
+        insertBeforeFirstReference(tabs, tab)
+    }
+
+    @CompileStatic
+    private static boolean contains(Object obj, Object item) {
+        obj instanceof Collection ? ((Collection) obj).contains(item) : obj == item
+    }
+
+    @CompileStatic
+    private static void insertBeforeFirstReference(List<Map<String, Object>> tabs, Map<String, Object> tab) {
+        for (def it = tabs.listIterator(); it.hasNext();) {
+            def currentTab = it.next()
+            if (contains(currentTab.insertBefore, tab.id)) {
```

Comment:
The only reason I used iterator and `LinkedList` instead of `findIndexOf` and `add(index, obj)` is a performance. `LinkedList` used to avoid expensive array movements, because we can often insert element into beginning or middle of list. But downside of LinkedList is slow access by index, so it should twice iterate list to insert new element: first time when looking matching item and second time when finding element by index for insertion.